### PR TITLE
📖 Add side-by-side comparison of MachinePools and MachineDeployments in docs

### LINKED ISSUE #4288: Add a side by side comparison of MachinePools and MachineDeployments in book
Maybe pictures would tell the story our words have failed to tell.

Documentation showing a visual representation comparing a `MachineDeployment` and `MachinePool` would be super helpful. Perhaps, a multi-step visualization starting from initial apply and ending with deployment completed with multiple machines that describes the anatomy, relationships and choreography of the resources the user specified and the ones created by the controller. This documentation could be less developer focused and more user focused; less state diagrams and more context / intent.

@felipeweb do you think that would have helped you?

_Originally posted by @devigned in https://github.com/kubernetes-sigs/cluster-api/pull/4112#discussion_r575659891_

### PR DESCRIPTION
**Pull request by Jont828**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4288

### CODE REVIEW

**Comment by fabriziopandini** [comment]
Thanks! 
/lgtm

/cc @CecileRobertMichon @richardcase

File: docs/book/src/tasks/experimental-features/machine-pools.md [hunk 1]
```diff
@@ -25,3 +25,17 @@ More details on `MachinePool` can be found at:
...
+
+| MachinePools | MachineDeployments |
+| - | - |
+| Creates new instances through a single infrastructure resource like VMSS in Azure or Autoscaling Groups in AWS. | Creates new instances by creating new Machines, which create individual VM instances on the infra provider. |
+| Set of instances is orchestrated by the infrastructure provider. | Set of instances is orchestrated by Cluster API using a MachineSet. |
+| Each MachinePool corresponds 1:1 with an associated InfraMachinePool. | Each MachineDeployment includes a MachineSet, and for each replica, it creates a Machine and InfraMachine. |
+| Each MachinePool requires only a single BootstrapConfig. | Each MachineDeployment uses an InfraMachineTemplate and a BootstrapConfigTemplate, and each Machine requires a unique BootstrapConfig. |
+| Maintains a list of instances in the `providerIDList` field in the MachinePool spec. This list is populated based on the response from the infrastructure provider. | Maintains a list of instances through the Machine resources owned by the MachineSet. |
+
+<!-- TODO: should we add MachinePoolMachines? -->
```
Comment (killianmuldoon) [inline docs/book/src/tasks/experimental-features/machine-pools.md:41]: I don't think it's required here until the implementation is further along, but definitely a TODO for future.

**Review by killianmuldoon — APPROVED** [review]
/lgtm

File: docs/book/src/tasks/experimental-features/machine-pools.md [hunk 2]
```diff
@@ -25,3 +25,17 @@ More details on `MachinePool` can be found at:
 [MachinePool CAEP](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20190919-machinepool-api.md)
 
 For developer docs on the MachinePool controller, see [here](./../../developer/architecture/controllers/machine-pool.md).
+
+## MachinePools vs MachineDeployments
+
+Although MachinePools provide a similar feature to MachineDeployments, MachinePools do so by leveraging an InfraMachinePool which corresponds 1:1 with a resource like VMSS on Azure or Autoscaling Groups on AWS which we treat as a black box. When a MachinePool is scaled up, the InfraMachinePool scales itself up and populates its provider ID list based on the response from the infrastructure provider. On the other hand, a when a MachineDeployment is scaled up, new Machines are created which then create an individual InfraMachine, which corresponds to a VM in Azure.
```
Comment (CecileRobertMichon) [inline docs/book/src/tasks/experimental-features/machine-pools.md:31]: > which corresponds to a VM in Azure

this makes it sound like Machine is an Azure only thing, might be better to stay generic on the infra, or to say "for example"

File: docs/book/src/tasks/experimental-features/machine-pools.md (hunk 1 above)
Comment (CecileRobertMichon) [reply docs/book/src/tasks/experimental-features/machine-pools.md:41]: @mboersma FYI

**Review by CecileRobertMichon — COMMENTED** [review]
lgtm

@Jont828 can you please squash commits?

**Comment by Jont828** [comment]
Sure thing!

**Comment by CecileRobertMichon** [comment]
Thank you!

/lgtm
/approve

**Pull request merged**